blocks/autocompleters/index.js

```diff
@@ -39,103 +45,41 @@ import BlockIcon from '../block-icon';
+ * @property {?FnGetOptionKeywords} getOptionKeywords get the keywords for a given option.
+ * @property {FnRenderOptionLabel} getOptionLabel get the label for a given option.
  * @property {?FnAllowNode} allowNode filter the allowed text nodes in the autocomplete.
  * @property {?FnAllowContext} allowContext filter the context under which the autocomplete activates.
- * @property {FnOnSelect} onSelect
- */
-
-/**
- * Returns an "completer" definition for selecting from available blocks to replace the current one.
- * The definition can be understood by the Autocomplete component.
- *
- * @param {Function} onReplace  Callback to replace the current block.
- *
- * @return {Completer} Completer object used by the Autocomplete component.
+ * @property {FnGetOptionCompletion} getOptionCompletion get the completion associated with a given option.
  */
-export function blockAutocompleter( { onReplace } ) {
-	// Prioritize common category in block type options
-	const options = sortBy(
-		getBlockTypes(),
-		( { category } ) => 'common' !== category
-	).map( ( blockType ) => {
-		const { name, title, icon, keywords = [] } = blockType;
-		return {
-			value: name,
-			label: [
-				<BlockIcon key="icon" icon={ icon } />,
-				title,
-			],
-			keywords: [ ...keywords, title ],
-		};
-	} );
-
-	const getOptions = () => Promise.resolve( options );
-
-	const allowContext = ( before, after ) => {
-		return ! ( /\S/.test( before.toString() ) || /\S/.test( after.toString() ) );
-	};
-
-	const onSelect = ( blockName ) => {
-		onReplace( createBlock( blockName ) );
-	};
-
-	return {
-		className: 'blocks-autocompleters__block',
-		triggerPrefix: '/',
-		getOptions,
-		allowContext,
-		onSelect,
-	};
-}
-/**
- * Returns a "completer" definition for inserting a user mention.
- * The definition can be understood by the Autocomplete component.
- *
- * @return {Completer} Completer object used by the Autocomplete component.
- */
-export function userAutocompleter() {
-	const getOptions = () => {
-		return wp.apiRequest( { path: '/wp/v2/users' } ).then( ( users ) => {
-			return users.map( ( user ) => {
-				return {
-					value: user,
-					label: [
-						<img key="avatar" className="blocks-autocompleters__user-avatar" alt="" src={ user.avatar_urls[ 24 ] } />,
-						<span key="name" className="blocks-autocompleters__user-name">{ user.name }</span>,
-						<span key="slug" className="blocks-autocompleters__user-slug">{ user.slug }</span>,
-					],
-					keywords: [ user.slug, user.name ],
-				};
-			} );
-		} );
-	};
-
-	const allowNode = () => {
-		return true;
-	};
 
-	const onSelect = ( user ) => {
-		return ( '@' + user.slug );
-	};
+export { default as blockAutocompleter } from './block';
+export { userAutocompleter };
 
-	return {
-		className: 'blocks-autocompleters__user',
-		triggerPrefix: '@',
-		getOptions,
-		allowNode,
-		onSelect,
-	};
-}
+export default [ userAutocompleter ];
```

Comment:
I'm not convinced that we should have the default setup exposed in here. We should rather set it in the hook.

Reply:
Agreed. I think that is a better idea. I'll make a note to make the update once we decide what to do instead of the `components.Autocomplete` filter.